### PR TITLE
Feature pageignore pattern for build

### PR DESCRIPTION
**Pull request by mindhivefi**
This is a suggested feature for making it possible to ignore files under pages -folder. This solution will add a new `pageIgnorePattern` -attribute into ´next.config.js´-file. When the value is set, page scanning will ignore files matching the given RegExp -pattern. 

This is a widely debated issue should this feature exist (https://github.com/vercel/next.js/issues/8974, https://github.com/vercel/next.js/issues/10377 and here https://github.com/vercel/next.js/issues/2332) . I decided to participate in dialog by making an implementation for the feature. There are two sides in this story, from semantical perspective the pages folder is dedicated for pages that will have an URL counter pair in the application. At the same time, there are several software development teams that have convention to keep the unit test code close to the actual code. This has been proven to be a good convention especially for big projects and for people doing multiple projects at the time. So for these teams missing this feature makes it quite hard to follow the convention used elsewhere. 

This feature is a small change for the code base and if you just don't use the configuration attribute, it does no harm for anyone. At the same time, for teams like us using tests close to code, it helps a lot.

### CODE REVIEW

**Comment by GrantGryczan** [comment]
What's the status on an RFC for this? This feature would be extremely useful to me.

**Comment by Cauen** [comment]
For people that's searching for a way to ignore files inside `/pages` folder
https://github.com/vercel/next.js/issues/3728#issuecomment-523789071
https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions
